src/automata/fsm.cljc

```diff
@@ -50,9 +50,10 @@
   [rules]
   (let [valid-states (set (keys rules))]
     (-> (into #{}
-              (comp (mapcat (comp ::to val))
+              (comp (mapcat val)
+                    (map ::to)
                     (distinct)
-                    (remove #(contains? valid-states %)))
+                    (remove valid-states))
```

Comment:
I'm a bit against directly relying on the fact that sets implement **IFn**, let's go with  `(partial contains? valid-states)` to make it easier on future readers

Reply:
ok, I switched to `(partial contains? valid-states)` in 9a8b0b9